src/celeritas/global/alongstep/detail/UniformFieldPropagatorFactory.hh

```diff
@@ -22,6 +22,11 @@ namespace detail
  */
 struct UniformFieldPropagatorFactory
 {
+#if CELER_USE_DEVICE
```

Comment:
We may need to make these device-dependent macros.

Reply:
what do you mean by device-dependant macros?

Reply:
I was thinking like CUDA A100 gets one setting, AMD MI250 gets another, etc.

Reply:
I suppose we can do that in subsequent PRs. I'm not even sure we want to keep the launch bounds in `UniformFieldPropagatorFactory`.

Reply:
Agreed. Oh yeah, the other thing I forgot is that for geometry-dependent functions like this one, we're going to further need *configuration*-dependent specifications (i.e. when running vecgeom on the A100, ...)